romancal/datamodels/tests/test_models.py

```diff
@@ -1,16 +1,186 @@
+import pytest
+
 import asdf
+import numpy as np
+from astropy.time import Time
+from copy import deepcopy
 
 from romancal.datamodels import RomanDataModel
+import romancal.datamodels as DataModels
+from stdatamodels.validate import ValidationWarning
 
+# Helper class to iterate over model subclasses
+def iter_subclasses(model_class):
+    yield model_class
+    for sub_class in model_class.__subclasses__():
+        yield from iter_subclasses(sub_class)
 
-def test_model_schemas():
-    def iter_subclasses(model_class):
-        yield model_class
-        for sub_class in model_class.__subclasses__():
-            yield from iter_subclasses(sub_class)
-
+def test_model_schemas(tmp_path):
     for model_class in iter_subclasses(RomanDataModel):
         try:
             asdf.schema.load_schema(model_class.schema_url)
         except Exception:
-            assert False, f"Unable to load schema URI '{model_class.schema_url}' for model class {model_class.__name__}"
+            assert False, f"Unable to load schema URI '{model_class.schema_url}' for model " \
+                          f"class {model_class.__name__}"
+
+
+
+# Helper function to test required keywords
+# meta = dictionary of required key/value pairs to ensure each are required
+def confirm_required_keywords(tmp_path, meta):
```

Comment:
I think you may be able to do a lot of this with existing `DataModel` methods:

```python
dm = RomanDataModel(meta)
for required_key in dm.keys():
  original_value = dm[required_key]
  dm[required_key] = None
  with pytest.warns(ValidationWarning):
    dm.validate()
  dm[required_key] = original_value
```
- DataModel.keys returns an iterator of all paths in the model (like "meta.instrument.name")
- Setting the value of a key to `None` is equivalent to removing it
- There is no need to check the length of the captured warnings, since `pytest.warns(ValidationWarning)` already asserts that the warning occurs at least once

Reply:
Replaced and records removed.